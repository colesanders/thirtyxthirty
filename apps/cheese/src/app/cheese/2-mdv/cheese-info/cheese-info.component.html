<p>cheese-info works!</p>
