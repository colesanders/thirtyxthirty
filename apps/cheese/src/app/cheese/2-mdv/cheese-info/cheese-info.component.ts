import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'thirty-cheese-info',
  templateUrl: './cheese-info.component.html',
  styleUrls: ['./cheese-info.component.scss']
})
export class CheeseInfoComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
