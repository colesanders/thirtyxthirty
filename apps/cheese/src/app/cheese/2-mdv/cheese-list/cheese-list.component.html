<p>cheese-list works!</p>
