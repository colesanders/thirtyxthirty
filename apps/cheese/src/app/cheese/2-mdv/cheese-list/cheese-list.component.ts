import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'thirty-cheese-list',
  templateUrl: './cheese-list.component.html',
  styleUrls: ['./cheese-list.component.scss']
})
export class CheeseListComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
