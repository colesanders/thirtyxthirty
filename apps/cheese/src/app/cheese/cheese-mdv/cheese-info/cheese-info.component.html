<mat-card class="mat-card-dark"> 
    <mat-card-header style="display: flex; justify-content: space-between;">
      <mat-card-title class="accent">
        <span *ngIf="cheeseForm.value.id !== null; else elseBlock">
          Editing Cheese
        </span>
        <ng-template #elseBlock>
          Create Cheese
        </ng-template>
      </mat-card-title>
      
    </mat-card-header>
  <form [formGroup]="cheeseForm" (ngSubmit)="saved.emit()">
    <mat-card-content>

      <mat-form-field class="full-width" style="width: 50%" >
        <mat-label *ngIf="cheeseForm.invalid && cheeseForm.touched">
          Name must contain at least 2 characters.
        </mat-label>
        <input  matInput required placeholder="Name" type="string"
        formControlName="name">
      </mat-form-field>
      <br>

      <mat-form-field class="full-width">
        <mat-label *ngIf="cheeseForm.invalid && cheeseForm.touched">
          Description must contain 10 characters.
        </mat-label>

        <textarea  matInput required placeholder="Description" type="text"
        formControlName="description">
        </textarea>
      </mat-form-field>
      <br>

      <mat-form-field class="full-width">
        <mat-label >
          Cost: {{cheeseForm.value.price | currency}}
        </mat-label>

        <input  matInput required placeholder="Price" type="number"
        formControlName="price"/>
      </mat-form-field>
      <br>


    </mat-card-content>
    <mat-card-actions style="display: flex; justify-content: space-between;">
      <button [disabled]="cheeseForm.invalid" type="submit" mat-button color="primary">Save</button>
      <button type="button" mat-button (click)="cancelled.emit();cheeseForm.reset();">
        Cancel
      </button>
    </mat-card-actions>
  </form>
</mat-card>