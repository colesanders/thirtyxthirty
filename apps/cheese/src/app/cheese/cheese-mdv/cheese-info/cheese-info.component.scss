@import '~@angular/material/theming';

mat-form-field {
    width: 100%;
}

mat-slider {
    width: 100%;
    margin: auto;
}

.star-color {
    color: #212bf3;
}

h4 {
    text-align: center; 
    color: mediumblue;
}

mat-icon {
    padding-top: 8px;
}

mat-radio-button {
    padding-right: 5px;
}