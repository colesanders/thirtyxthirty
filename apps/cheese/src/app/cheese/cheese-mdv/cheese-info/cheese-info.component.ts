import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Cheese } from '@thirty/api-interfaces';


@Component({
  selector: 'thirty-cheese-info',
  templateUrl: './cheese-info.component.html',
  styleUrls: ['./cheese-info.component.scss']
})
export class CheeseInfoComponent implements OnInit {
  @Input() cheeseForm;
  @Output() saved = new EventEmitter();
  @Output() cancelled = new EventEmitter();

  dropDisplay: number[];
  

  constructor() { }

  ngOnInit(): void {
    this.dropDisplay = [1,1,1,1,1];
  }
}
