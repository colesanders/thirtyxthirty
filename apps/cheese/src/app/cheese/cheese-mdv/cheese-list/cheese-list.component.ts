import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Cheese } from '@thirty/api-interfaces';

@Component({
  selector: 'thirty-cheese-list',
  templateUrl: './cheese-list.component.html',
  styleUrls: ['./cheese-list.component.scss']
})
export class CheeseListComponent implements OnInit {
  @Input() cheeses: [Cheese];
  @Output() selected = new EventEmitter<Cheese>();
  @Output() deleted = new EventEmitter<Cheese>();
  constructor() { }

  ngOnInit(): void {
  }

}
