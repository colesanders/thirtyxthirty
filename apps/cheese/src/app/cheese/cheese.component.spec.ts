import { HttpClientModule } from '@angular/common/http';
import { DebugElement } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { CheeseService } from '@thirty/core-data';
import { MaterialModule } from '@thirty/material';
import { of } from 'rxjs';
import { CheeseComponent } from './cheese.component';
import { Cheese } from '@thirty/api-interfaces';
import { CheeseInfoComponent } from './cheese-mdv/cheese-info/cheese-info.component';
import { CheeseListComponent } from './cheese-mdv/cheese-list/cheese-list.component';

const mockEmptyCheese: Cheese = {
  id: null,
  name: 'Mock Empty Cheese',
  description: '',
  tags: '',
  rating: '',
  price: 0,
};

const mockCheese : Cheese = {
  id: 1,
  name: 'Mock Cheese',
  description: 'Mock',
  tags: '',
  rating: '',
  price: 0,
};

const mockCheeseService = {
  getCheeses: () => of([]),
  createCheese: () => of({}),
  updateCheese: () => of({}),
  deleteCheese: () => of({})
}

describe('CheeseComponent', () => {
  let component: CheeseComponent;
  let fixture: ComponentFixture<CheeseComponent>;
  let de: DebugElement;
  let cheeseService: CheeseService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        HttpClientModule,
        MaterialModule,
        NoopAnimationsModule,
        RouterTestingModule,
      ],
      providers: [
        { provide: CheeseService, useValue: mockCheeseService }
      ],
      declarations: [
        CheeseComponent,
        CheeseInfoComponent,
        CheeseListComponent
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CheeseComponent);
    component = fixture.componentInstance;
    de = fixture.debugElement;
    cheeseService = de.injector.get(CheeseService);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should set route as detail 1', () => {
    component.updateLinks(1);
    expect(component.links[1].path).toMatchSnapshot();
  });
});
