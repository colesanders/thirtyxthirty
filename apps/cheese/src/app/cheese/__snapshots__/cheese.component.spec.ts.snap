// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheeseComponent should select a cheese 1`] = `
Array [
  "detail/",
  1,
]
`;
